### DISCUSSION TITLE
I made an AMD ROCm6.2 compatible docker file for AMD GPU users!

### DISCUSSION THREAD

**Discussion by benhaotang** [adithya-s-k/omniparse#99] [Show and tell]
# TL;DR

for all the torch ROCm6.2 supported GPUs (MI200,MI300,7900XT,7800XT,6900XT,etc.)
```
docker run -it -p 8000:8000 --device=/dev/kfd --device=/dev/dri --group-add=video --ipc=host --cap-add=SYS_PTRACE --security-opt seccomp=unconfined benhaotang/omniparse-rocm:v0.1
```
you should have everything running without any problem (I tested on Ubuntu 24.04 with my 7800XT)

# Full method
1. I pulled the ROCm offical docker file
```
 docker run -it --network=host --device=/dev/kfd --device=/dev/dri --group-add=video --ipc=host --cap-add=SYS_PTRACE --security-opt seccomp=unconfined rocm/pytorch:rocm6.2.3_ubuntu22.04_py3.10_pytorch_release_2.3.0
```
2. cloned this repo and rocm/transformers
```
git clone https://github.com/adithya-s-k/omniparse.git
git clone https://github.com/ROCm/transformers.git
```
3. build transformers-rocm with `python setup.py install`
 
4. force install dependency version to not overwrite torch-rocm and scikit-learn-rocm **Please always make sure torch torchvision and scikit learn are the original version from the rocm docker hub!!!! Never let other dependency overwrite it**
```
pip install surya-ocr==0.4.15 marker-pdf==0.2.16 --no-dependeces
```
5. modify pyproject.toml to remove the requirements of torch, flash_attn surya-ocr marker-pdf, and just install everything normally
```
pip install -e .
```
It will complain "pip cannot consider all conflicts",  just ignore the warning (because the torch-rocm in the docker image is an alpha build of 2.3.0, pip has problem resolving it.

6. install chrome
7. `python server.py --host 0.0.0.0 --port 8000 --documents --media --web` should run without any problem

# Last note

At current stage, I really suggest anyone running an AMD GPU to use the docker method instead of venv, docker files from ROCm has the most up-to-date but compatible dependencies, but to solve dependency problems in stable driver is just a mess. only downside is that the ROCm docker image is 56GB... so what ever you do, be prepared for a large sum of storage consumption.

I will also try to make a dockerfile and automate the pushing process with main repo when I am free recently.
